File: src/main/java/ru/citeck/ecos/process/domain/proc/repo/ProcStateRepository.java
package ru.citeck.ecos.process.domain.proc.repo;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;
import ru.citeck.ecos.process.domain.common.repo.EntityUuid;
import ru.citeck.ecos.process.domain.proc.entity.ProcessStateEntity;

@Repository
public interface ProcStateRepository extends MongoRepository<ProcessStateEntity, EntityUuid> {

}

File: src/main/java/ru/citeck/ecos/process/domain/proc/repo/ProcStateRepository.kt
package ru.citeck.ecos.process.domain.proc.repo

import ru.citeck.ecos.process.domain.common.repo.EntityUuid
import ru.citeck.ecos.process.domain.procdef.repo.ProcDefRevEntity

interface ProcStateRepository {

    fun findById(id: EntityUuid): ProcessStateEntity?

    fun save(entity: ProcessStateEntity): ProcessStateEntity

    fun findFirstByProcDefRevIn(procDefRev: List<ProcDefRevEntity>): ProcessStateEntity?
}
